### ISSUE TITLE
Public folder visible on the URL

### ISSUE THREAD

**Issue by FlavioSuarez** [abydahana/aksara#582]
Hi,

Started 2 fresh installs of Aksara 5.1.6 version (one in basic mode and other in developer mode) with the current CI version (4.5.5).
In both installs the /public/ folder appears on the URL.
Changing the 'ENVIRONMENT' setting has no effect.

I have a 5.1.1 version installed (with the 'ENVIRONMENT' setted to 'development') with CI 4.5.3 and the '/public/' folder don't appears on the URL...

Am I missing something?

TIA and keep the good work!!

**Comment by abydahana** [comment]
You need to point your sub/domain to /path/to/aksaracms/public directly or follow this step instead: https://github.com/abydahana/aksara/discussions/439

**Comment by FlavioSuarez** [comment]
Hi,

I'm in my local machine using Laragon (same as XAMPP) as development environment. The same Laragon on wich I run the older version working ok...
The CI's BaseURL config seems ok: "http://aksara-main.test/public/"
The routes seems ok, too.
So, it seems it's nothing related with your advices...

I did a fresh install of the 5.1.1 released version here at home (wich Composer downloaded the current CI 4.5.5) and the "/public/" appears on the URL too.
As I said, the ok version (at work) runs on CI 4.5.3. Could be the CI version the reason of different behavior?
I'll do more tests on monday at work...

TIA. Best regards.

**Comment by mdestafadilah** [comment]
> Hi,
> 
> Started 2 fresh installs of Aksara 5.1.6 version (one in basic mode and other in developer mode) with the current CI version (4.5.5). In both installs the /public/ folder appears on the URL. Changing the 'ENVIRONMENT' setting has no effect.
> 
> I have a 5.1.1 version installed (with the 'ENVIRONMENT' setted to 'development') with CI 4.5.3 and the '/public/' folder don't appears on the URL...
> 
> Am I missing something?
> 
> TIA and keep the good work!!

what's your server using? nginx, apacher, litespeed or others?

**Comment by abydahana** [comment]
I tested both Apache and Nginx

**Comment by FlavioSuarez** [comment]
Apache 2.4.59-win64-VS17
PHP 8.3.8-x64 [TS]
MySQL 8.4.0

I did a fresh install of 5.1.1 release, developer mode (without sample) and CI 4.5.5, here at work and the /public/ folder appears on the URL too...
I downloaded the 5.1.1 release, unziped it into an empty folder and ran composer install wich downloaded CI 4.5.5. When I open the new site and start the install process, the /public/ folder already appears.

The ok 5.1.1 (on CI 4.5.3) version was formerly a 5.0.6 version wich was updated to 5.1.0 and after to 5.1.1, by copying the new released files over the priors. 
Also, the app Description don't appears on the footer of the ok version, while it does on the NOT OK version...

Later I'll try to do a fresh install forcing the CI 4.5.3 and see what happens...

**Comment by FlavioSuarez** [comment]
More info:

On the CI's debugbar, I can see that BaseURL is setted to "http://aksara-main.test/public/" instead of "http://aksara-main.test".
Couldn't find where this '/public/' is appended to the $baseURL...

**Comment by abydahana** [comment]
> More info:
> 
> On the CI's debugbar, I can see that BaseURL is setted to "http://aksara-main.test/public/" instead of "http://aksara-main.test". Couldn't find where this '/public/' is appended to the $baseURL...

Could you please share your virtual host config?

**Comment by FlavioSuarez** [comment]
This is the content created by Laragon on the hosts file:

127.0.0.1      aksara-511.test      #laragon magic!   
127.0.0.1      aksara-main.test     #laragon magic!   
127.0.0.1      aksaracms.test       #laragon magic!

**Comment by mdestafadilah** [comment]
> This is the content created by Laragon on the hosts file:
> 
> 127.0.0.1 aksara-511.test #laragon magic! 127.0.0.1 aksara-main.test #laragon magic! 127.0.0.1 aksaracms.test #laragon magic!

i'm using laragon too, but not try on localhost, may you can use this

```bash
<VirtualHost *:80>
    ServerAdmin webmaster@localhost
    ServerName ahntest.net
    DocumentRoot /home/www/ahntest.net

    Alias /testberg /home/www/testberg
    <Directory "/home/www/testberg">
        Options +Indexes
        AllowOverride All
    </Directory>
</VirtualHost>
```

// source: https://serverfault.com/a/347549

hope this help you little.

**Comment by abydahana** [comment]
> > This is the content created by Laragon on the hosts file:
> > 127.0.0.1 aksara-511.test #laragon magic! 127.0.0.1 aksara-main.test #laragon magic! 127.0.0.1 aksaracms.test #laragon magic!
> 
> i'm using laragon too, but not try on localhost, may you can use this
> 
> ```shell
> <VirtualHost *:80>
>     ServerAdmin webmaster@localhost
>     ServerName ahntest.net
>     DocumentRoot /home/www/ahntest.net
> 
>     Alias /testberg /home/www/testberg
>     <Directory "/home/www/testberg">
>         Options +Indexes
>         AllowOverride All
>     </Directory>
> </VirtualHost>
> ```
> 
> // source: https://serverfault.com/a/347549
> 
> hope this help you little.

Hi, please check the virtual host config under your nginx and make sure the "`root`" parameter is pointed to /public directly:
<img width="1440" alt="image" src="https://github.com/user-attachments/assets/57544b4c-5ef0-4033-97e9-08e01e218868">

The final config should be like this:
```
server {
    listen 80;
    server_name aksara-main.test *.aksara-main.test;
    root "/path/to/aksaracms/public"; # LOOK AT THIS
    
    index index.html index.htm index.php;
 
    location / {
        try_files $uri $uri/ /index.php$is_args$args;
		autoindex on;
    }
    
    location ~ \.php$ {
        include snippets/fastcgi-php.conf;
        fastcgi_pass php_upstream;		
        #fastcgi_pass unix:/run/php/php7.0-fpm.sock;
    }
	
	
    charset utf-8;
	
    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }
    location ~ /\.ht {
        deny all;
    }
}

# This file is auto-generated.
# If you want Laragon to respect your changes, just remove the [auto.] prefix
# If you want to use SSL, enable it at: Menu > Nginx > SSL > Enabled
```

**Comment by FlavioSuarez** [comment]
You're using Laragon 5.0.0. Mine is 6.0.0.
Nginx is disabled in my Preferences. I use only Apache. Ever.
I know that Aksara don't needs the .env file. I use it on others CI projects, setting the BaseURL as I said and database credentials.
Opening the CI's DebugBar I can see that the BaseURL is setted without the '/public/' on the updated Aksara 5.1.1. But on the fresh installs the BaseURL is setted **with** the '/public/'.
The defined BASE_URL on the config.php seems ok. 
So, somewhere, sometime, the public $baseURL variable is getting the '/public/' added to the end of the URL..

![Captura de tela 2024-10-01 123002](https://github.com/user-attachments/assets/30c47b3c-524f-4d16-8567-03b90060e242)

![Captura de tela 2024-10-01 122802](https://github.com/user-attachments/assets/76485a7d-ae9a-4568-a589-fa99b0d34a52)

![Captura de tela 2024-10-01 123210](https://github.com/user-attachments/assets/69703827-6fd5-4a87-bd02-79a3ff703b65)

Thanks for your time with this issue... Hope we can find the answer to this mistery... :-)

**Comment by mdestafadilah** [comment]
> Laragon is using Nginx by default. You can check the screenshot on my previous reply just to make sure the virtual host are created correctly.
> 
> I tested with Laragon also and had no issue.
> 
> And there's no .env required in Aksara's project, so I wonder why you put .env in yours 😁

Sir, we can choose nginx or apache as webservice ...
![02-10-2024 08-44-10](https://github.com/user-attachments/assets/80f7b1c6-a3c6-4ccd-bd18-c424d60812d9)



 but this is very mistery ... can you downgrade your laragon version sir @FlavioSuarez only for testing...

i'm using laragon 4 but with apache and php newest.

**Comment by FlavioSuarez** [comment]
Bingo!!! Found it!!!
Now I see what you both mean with

>
> Could you please share your virtual host config?
>

> ```shell
> <VirtualHost *:80>
>     ServerAdmin webmaster@localhost
>     ServerName ahntest.net
>     DocumentRoot /home/www/ahntest.net
> 
>     Alias /testberg /home/www/testberg
>     <Directory "/home/www/testberg">
>         Options +Indexes
>         AllowOverride All
>     </Directory>
> </VirtualHost>
> ```

I was looking for in the "hosts" file.
But right now I had an insight and found these settings on the Apache > sites-enabled > *.conf files
On the working site, the "DocumentRoot" and "Directory" have "/public" on the path.
On the NOT working site, they doesn't.

Then the 2nd insight was: The ok site was created by running a Composer installation, while in the not-ok sites I did a manual installation.
In the manual installation I first created a blank project on Laragon by going to Quick App > Blank (wich created the folder, the database **and** the .conf file **without** the "/public" on the "DocumentRoot" and "Directory" settings) and **then** I did the manual installation into this folder.
Tested right now doing a Composer installation and all seems ok again!!

Thank you so much for your help, guys!!
Re-starting my testing project from zero...
Keep the good work! This project is really awesome!

![image](https://github.com/user-attachments/assets/fc05d97f-44ae-4e27-b390-7eded5ce69df)

![image](https://github.com/user-attachments/assets/a6fdb0d6-380a-440f-af35-d596e09b40de)
